### ISSUE TITLE
What licence is the code in this repo?

### ISSUE THREAD

**Issue by Lazerbeak12345** [minetest-mods/vehicle_mash#10]
This... is a hairy licencing issue.

- It looks like the things from "Car" by Esteban are under CC-BY-NC-SA. (code included, I assume?)  -- EDIT code is MIT
- "Mesecars" by paramat and "Boats" by PilzAdam code WTFPL
- "Hovercraft" by Stuart Jones doesn't say what the code is under. (it specifies every other detail)

But what about the code in `./framework.lua`? What about other code not from the above?

## Todos

Since there is some work to do here apparently:

- [ ] indicate that the code from Car (if any) is MIT
- [ ] (somehow) clarify what licence new code falls under. (WTFPL licence)

**Comment by Panquesito7** [comment]
Good catch. Actually, the `Car` textures from Esteban are under CC BY-NC-SA, but that's *only* the textures, there's no code included for that. The code itself should be MIT.

Hopefully, @blert2112 knows more about this. Thank you. 🙂

**Comment by Lazerbeak12345** [comment]
I think what this mod does at the bottom of the README is a great solution: https://github.com/minetest-streets/streets but you must be sure it's accurate. (that is - I like that there's a catchall. It means that all new code modifications are covered)

**Comment by Lazerbeak12345** [comment]
upon some research the [hovercrafts mod](https://content.minetest.net/packages/AntumDeluge/hovercraft_ad/) is _currently_ under the LGPL 3.0 or later. Keep in mind - that's the current licence, not necessarily what it was under when @blert2112 forked it into this mod.

**Comment by Panquesito7** [comment]
That's a fork of the `hovercrafts` mod. The license details for the original one can be found here https://github.com/stujones11/hovercraft/blob/master/LICENSE.txt

**Comment by Lazerbeak12345** [comment]
I see. The original is LGPL. Was it under that licence when vehicle_mash forked it?

**Comment by blert2112** [comment]
Let me see of I can clarify... if I can recall correctly.
There is no code from all these other vehicle mods. It's is only the textures/models. I tried to link their relevant threads when I could. The code of the mod itself is basically just a registering type thing. Something to say "hey, I exist and these are my values", lib_mount handles everything else. TBH, I probably got this idea straight from mobs_redo as I was working on features for that at the same time. I did this because when my server was running I found that there was a crap ton of duplicated code throughout many of the mods I was using. Vehicle_mash and lib_mount allow you to take whatever models and textures you want and, given the correct values for registration, use them without all the code duplication. Pretty sure at one point I even had the dragon working but I may have lost that section of the project years ago. Hope that clears it up some.

To sum up... there was no Forking of the various mods vehicles came from (which, BTW, were all just forks of  PilzAdam's original boats mod) just the textures/models.

**Comment by blert2112** [comment]
The license info was correct and as up to date as possible when I originally wrote the readme.

**Comment by Lazerbeak12345** [comment]
Sounds like the licence for new code changes should be that of the Boats mod.

**Comment by Panquesito7** [comment]
This issue is very old, but, in this case, the license for the code would be WTFPL.
I suggest we should change it to MIT or the most appropriate license. I'm not sure if we need to ask the contributors for this, though.
